### ISSUE TITLE
no-unused-declaration: doesn't work with class extends

### ISSUE THREAD

**Issue by Hotell** [cartant/eslint-plugin-etc#7]
Found another bug I guess. Thx!

**version:** eslint-plugin-etc@0.0.1-beta.14
**ESlint config:**

```js
'etc/no-unused-declaration': ['error', { ignored: { React: true } }],
```

![image](https://user-images.githubusercontent.com/1223799/66553741-f52dba00-eb4b-11e9-901b-86f87730dcdc.png)

**Comment by cartant** [comment]
Thanks. Can you give me a snippet that makes clear what the problem is so that I can create a failing test? I can't see what's going on in the image, but I suppose it could be this phone.

**Comment by Hotell** [comment]
sure thingy

```ts
// ❌ this is a lint error, although it's used as a base class
import RestCollectionStore from 'our/custom/flux'

import {SpaceMembership} from './models'

class SpaceMembershipStore extends RestCollectionStore<SpaceMembership> {
// ..implementation
 }

```

**Comment by cartant** [comment]
Should be fixed in `0.0.1-beta.15`.

**Comment by Hotell** [comment]
that change broke jsx again it seems :(

**Comment by Hotell** [comment]
![image](https://user-images.githubusercontent.com/1223799/66584308-2d9dba00-eb85-11e9-9d64-ddcdb3c2fd1b.png)

**Comment by cartant** [comment]
I don't see how that's possible. The change adds a check **after** JSX usage has already been checked:

https://github.com/cartant/eslint-plugin-etc/commit/a3efeb6d41c7a691f4eafe6d272669c8b9aed2e6

There was another refactoring change in that release, but I cannot see how that could have broken anything:
https://github.com/cartant/eslint-plugin-etc/commit/8ab3679e203e7c3b554de51aa727e3044119c6f6

Also, the test that use JSX passes.

This leads me to think that something else could be happening. Does it effect a lint failure for this sort of thing in all of your files or just this one? Is there something different with this file? Also, does the previous version of the beta not effect the failure?

ATM, I don't really have enough information to efficiently solve the problem. Up until now these bugs have been really easy to fix.

**Comment by cartant** [comment]
This change did not effect a reversion. TypeScript uses different AST nodes for opening and self-closing JSX tags and the previous JSX-related PR checked only for the former.

Fixed in `0.0.1-beta.16`.

**Comment by Hotell** [comment]
Will do ! Thx a lot 🍻

**Comment by cartant** [comment]
I remembered the two-different-JSX-AST-node thing when I was in the shower. Showers are so useful for solving problems. 🚿

**Comment by Hotell** [comment]
Haha oh yeah, helps all the time 👌
